Scripts/build.swift

```diff
@@ -20,38 +20,132 @@ enum TaskError: Error {
     case code(Int32)
 }
 
-guard CommandLine.arguments.count > 3 else {
-    print("Usage: build.swift sdk destination should_test? enable_code_coverage?")
+enum Platform: String, CaseIterable, CustomStringConvertible {
+    case iOS_12
+    case iOS_13
+    case tvOS_12
+    case tvOS_13
+    case macOS_10_15
+    case watchOS_5
+    case watchOS_6
+
+    var destination: String {
+        switch self {
+        case .iOS_12:
+            return "platform=iOS Simulator,OS=12.2,name=iPad Pro (12.9-inch) (3rd generation)"
+        case .iOS_13:
+            return "platform=iOS Simulator,OS=13.2.2,name=iPad Pro (12.9-inch) (3rd generation)"
+
+        case .tvOS_12:
+            return "platform=tvOS Simulator,OS=12.2,name=Apple TV"
+        case .tvOS_13:
+            return "platform=tvOS Simulator,OS=13.2,name=Apple TV"
+
+        case .macOS_10_15:
+            return "platform=OS X"
+
+        case .watchOS_5:
+             return "OS=5.2,name=Apple Watch Series 4 - 44mm"
+        case .watchOS_6:
+            return "OS=6.1,name=Apple Watch Series 4 - 44mm"
+        }
+    }
+
+    var sdk: String {
+        switch self {
+        case .iOS_12,
+             .iOS_13:
+            return "iphonesimulator"
+
+        case .tvOS_12,
+             .tvOS_13:
+            return "appletvsimulator"
+
+        case .macOS_10_15:
+            return "macosx10.15"
+
+        case .watchOS_5,
+             .watchOS_6:
+            return "watchsimulator"
+        }
+    }
+
+    var shouldTest: Bool {
+        switch self {
+        case .iOS_12,
+             .iOS_13,
+             .tvOS_12,
+             .tvOS_13:
+            return true
+
+        case .macOS_10_15:
+            // We can't run tests on macOS 10.15 because CacheAdvance assumes macOS 10.15 APIs are available, and Travis CI's machines run on macOS 10.14.
+            // For more information, see https://github.com/dfed/CacheAdvance/issues/2
+            return false
+
+        case .watchOS_5,
+             .watchOS_6:
+            // watchOS does not support unit testing (yet?).
+            return false
+        }
+    }
+
+    var enableCodeCoverage: Bool {
```

Comment:
A comment here would be great re: why we're only gathering code coverage for tv OS

Reply:
I'll add something, though the reasoning is "there was no need to do more than this" 😅